packages/nvidia_nat_core/src/nat/runtime/session.py

```
@@ -384,6 +387,79 @@ def _get_user_id_from_context(self) -> str | None:
+        """
+        if not token or token.count(".") != 2:
+            return None
+        try:
+            payload_b64 = token.split(".")[1]
+            padding = 4 - len(payload_b64) % 4
+            if padding != 4:
+                payload_b64 += "=" * padding
+            payload_bytes = base64.urlsafe_b64decode(payload_b64)
+            out = json.loads(payload_bytes.decode("utf-8"))
+            return out if isinstance(out, dict) else None
+        except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)